Jellyfin.Server/Migrations/Routines/FixAudioData.cs

```diff
@@ -0,0 +1,104 @@
+using System.Threading;
+using Jellyfin.Data.Enums;
+using MediaBrowser.Controller;
+using MediaBrowser.Controller.Entities;
+using MediaBrowser.Controller.Entities.Audio;
+using MediaBrowser.Controller.Persistence;
+using MediaBrowser.Model.Entities;
+using Microsoft.Extensions.Logging;
+
+namespace Jellyfin.Server.Migrations.Routines
+{
+    /// <summary>
+    /// Fixes the data column of audio types to be deserializable.
+    /// </summary>
+    internal class FixAudioData : IMigrationRoutine
+    {
+        private const string DbFilename = "library.db";
+        private readonly ILogger<FixAudioData> _logger;
+        private readonly IServerApplicationPaths _applicationPaths;
+        private readonly IItemRepository _itemRepository;
+
+        public FixAudioData(
+            IServerApplicationPaths applicationPaths,
+            ILoggerFactory loggerFactory,
+            IItemRepository itemRepository)
+        {
+            _applicationPaths = applicationPaths;
+            _itemRepository = itemRepository;
+            _logger = loggerFactory.CreateLogger<FixAudioData>();
+        }
+
+        /// <inheritdoc/>
+        public Guid Id => Guid.Parse("{CF6FABC2-9FBE-4933-84A5-FFE52EF22A58}");
+
+        /// <inheritdoc/>
+        public string Name => "FixAudioData";
+
+        /// <inheritdoc/>
+        public bool PerformOnNewInstall => false;
+
+        /// <inheritdoc/>
+        public void Perform()
+        {
+            var dbPath = Path.Combine(_applicationPaths.DataPath, DbFilename);
+
+            // Back up the database before modifying any entries
+            for (int i = 1; ; i++)
+            {
+                var bakPath = string.Format(CultureInfo.InvariantCulture, "{0}.bak{1}", dbPath, i);
+                if (!File.Exists(bakPath))
+                {
+                    try
+                    {
+                        File.Copy(dbPath, bakPath);
+                        _logger.LogInformation("Library database backed up to {BackupPath}", bakPath);
+                        break;
+                    }
+                    catch (Exception ex)
+                    {
+                        _logger.LogError(ex, "Cannot make a backup of {Library} at path {BackupPath}", DbFilename, bakPath);
+                        throw;
+                    }
+                }
+            }
+
+            _logger.LogInformation("Backfilling audio lyrics data to database.");
+            var startIndex = 0;
+            var records = _itemRepository.GetCount(new InternalItemsQuery
+            {
+                IncludeItemTypes = [BaseItemKind.Audio],
+            });
+
+            while (startIndex < records)
+            {
+                var results = _itemRepository.GetItemList(new InternalItemsQuery
+                {
+                    IncludeItemTypes = [BaseItemKind.Audio],
+                    StartIndex = startIndex,
+                    Limit = 100,
+                    SkipDeserialization = true
```

Comment:
What's the impact of not skipping deserialization given that before this change nothing would be serialized for Audio anyway?

All the changes to pass the skip flag just seem like dead code once this migration has run, so how valuable is it to add?

Reply:
Allowing deserialization means that any non-BaseItem field not annotated with @JsonIgnore is being saved as one big JSON blob into the database.
In this case it only includes the LyricFiles.
The migration alone is not enough because the database field would never get updated.

Reply:
Otherwise a simple rescan after enabling serialization would have been enough to fix the issue.